rust/sdk-processor/src/processors/events_processor.rs

```diff
@@ -0,0 +1,120 @@
+}
+
+impl EventsProcessor {
+    pub async fn new(config: IndexerProcessorConfig) -> Result<Self> {
+        let db_config = config.db_config.clone();
+        match db_config {
+            DbConfig::PostgresConfig(postgres_db_config) => {
+                let conn_pool = new_db_pool(
+                    &postgres_db_config.connection_string,
+                    Some(postgres_db_config.db_pool_size),
+                )
+                .await
+                .map_err(|e| {
+                    anyhow::anyhow!(
+                        "Failed to create connection pool for PostgresConfig: {:?}",
+                        e
+                    )
+                })?;
+
+                Ok(Self {
+                    config,
+                    postgres_config: postgres_db_config.clone(),
+                    db_pool: conn_pool,
+                })
+            },
+        }
+    }
+
+    pub async fn run_processor(self) -> Result<()> {
+        // (Optional) Run migrations
+        run_migrations(
+            self.postgres_config.connection_string.clone(),
+            self.db_pool.clone(),
+        )
+        .await;
+
+        // (Optional) Merge the starting version from config and the latest processed version from the DB
+        let starting_version = get_starting_version(&self.config, self.db_pool.clone()).await?;
+
+        // Define processor steps
+        let (_input_sender, input_receiver) = instrumented_bounded_channel("input", 1);
+
+        let transaction_stream = TransactionStreamStep::new(TransactionStreamConfig {
+            starting_version: Some(starting_version),
+            ..self.config.transaction_stream_config
+        })
+        .await?;
+        let transaction_stream_with_input = RunnableStepWithInputReceiver::new(
+            input_receiver,
+            transaction_stream.into_runnable_step(),
+        );
+        let events_extractor = EventsExtractor {};
+        let events_storer = EventsStorer::new(self.db_pool.clone(), self.postgres_config.clone());
+        let timed_buffer = TimedBuffer::new(Duration::from_secs(1));
+        let version_tracker = LatestVersionProcessedTracker::new(
+            self.db_pool.clone(),
+            starting_version,
+            self.config.processor_config.name().to_string(),
+        );
+
+        // Connect processor steps together
+        let (_, buffer_receiver) = ProcessorBuilder::new_with_runnable_input_receiver_first_step(
+            transaction_stream_with_input,
+        )
+        .connect_to(events_extractor.into_runnable_step(), 10)
+        .connect_to(timed_buffer.into_runnable_step(), 10)
+        .connect_to(events_storer.into_runnable_step(), 10)
+        .connect_to(version_tracker.into_runnable_step(), 10)
+        .end_and_return_output_receiver(10);
+
+        // (Optional) Parse the results
+        loop {
+            match buffer_receiver.recv().await {
+                Ok(txn_context) => {
+                    if txn_context.data.is_empty() {
```

Comment:
why do we even allow empty data? any chance we make it a warn! or error!?

Reply:
We don't process events for certain transaction types like block epilogue transactions. For transactions with no extracted data, the data would be empty and that is fine. But it still needs to be passed through the pipeline to track the processed versions.